docker/adaptivecpp/Dockerfile

```diff
@@ -14,15 +14,22 @@ RUN export DEBIAN_FRONTEND=noninteractive && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists*
 
-RUN git clone https://github.com/illuhad/hipSYCL.git \
-      --branch=develop --single-branch --shallow-since=2021-08-01 \
-      --recurse-submodules /tmp/hipSYCL && \
-    cd /tmp/hipSYCL && \
+RUN git clone https://github.com/AdaptiveCpp/AdaptiveCpp.git \
+      --branch=develop --single-branch --shallow-since=2024-12-01 \
+      --recurse-submodules /tmp/AdaptiveCpp && \
+    cd /tmp/AdaptiveCpp && \
     git checkout $IMPL_VERSION && \
-    cmake /tmp/hipSYCL -G Ninja -B /tmp/build \
+    cmake /tmp/AdaptiveCpp -G Ninja -B /tmp/build \
+      -DWITH_SSCP_COMPILER=OFF \
+      -DWITH_STDPAR_COMPILER=OFF \
+      -DWITH_ACCELERATED_CPU=OFF \
+      -DWITH_CUDA_BACKEND=OFF \
+      -DWITH_ROCM_BACKEND=OFF \
+      -DWITH_OPENCL_BACKEND=OFF \
```

Comment:
For #1012, we should decide for whether we want an LLVM-based build. In this context, I wonder whether there is any issue or challenge enabling the compiler features? 

Since AdaptiveCpp even when only targeting CPU now typically relies on LLVM (via `--acpp-targets=generic` by either JIT-compiling for OpenMP or OpenCL), we could use such a build without having to set up CUDA or ROCm and without needing GPU hardware.

As most users also are going to be using `--acpp-targets=generic`, that might be useful to have in CTS CI, if possible.

Reply:
@illuhad, I managed to install LLVM and successfully built default version of AdaptiveCpp. Unfortunately, I can't merge #1012 because SYCL-CTS code base relies on hipSYCL. I think this PR resolve the issue blocking #1012.
Could you please approve this PR and review #1012?

Reply:
I suggest to merge this PR first and then look into switching to SSCP in a follow up. It should only require changes to the Dockerfile (hopefully), and by merging this now we can get all the string replacements out of the way before they accrue too many conflicts with other PRs.

Reply:
Sounds good to me. @illuhad?

Reply:
Yep, okay for me!